src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -384,19 +383,13 @@ public function send()
      *
      * Valid types are strings, numbers, null, and objects that implement a __toString() method.
      *
-     * @param mixed $content Content that can be cast to string
-     *
      * @return $this
      *
      * @throws \UnexpectedValueException
      */
-    public function setContent($content)
+    public function setContent(?string $content)
     {
-        if (null !== $content && !\is_string($content) && !is_numeric($content) && !\is_callable([$content, '__toString'])) {
-            throw new \UnexpectedValueException(sprintf('The Response content must be a string or object implementing __toString(), "%s" given.', \gettype($content)));
```

Comment:
fully agree as the casting should be done in user-land when they need strict types compatibility.